.semaphore/semaphore.yml

```diff
@@ -500,6 +500,42 @@ blocks:
         value: latest
       commands:
       - "./support/bundler_wrapper exec rake test"
+    - name: Ruby 2.7.8 for redis-4
+      env_vars:
+      - *2
+      - *3
+      - *4
+      - *5
+      - name: RUBY_VERSION
+        value: 2.7.8
+      - name: GEMSET
+        value: redis-4
+      - name: BUNDLE_GEMFILE
+        value: gemfiles/redis-4.gemfile
+      - name: _RUBYGEMS_VERSION
+        value: latest
+      - name: _BUNDLER_VERSION
+        value: latest
+      commands:
+      - "./support/bundler_wrapper exec rake test"
+    - name: Ruby 2.7.8 for redis-5
```

Comment:
The build failure for this job seems quite consistent: https://appsignal.semaphoreci.com/jobs/d5fa2890-3c0c-4a57-9ef6-8eb7f1992292

Is there a known issue with hiredis on Ruby 2.7?

Reply:
Unfortunately, I haven't found anything about this, and I can't reproduce it either. The entire build matrix works perfectly on my local machine. So I'm a bit stuck here.

Reply:
Did some research on this. Long story short, there's something wrong with the OpenSSL extension in the build of Ruby 2.7.8 that Semaphore installs from its CDN. Manually re-building Ruby on the Semaphore instance directly (`rbenv install 2.7.8`) fixes the issue. A potential fix would be to do that and cache the resulting Ruby build across Semaphore job executions.

That said, the issue does not take place if we [upgrade the Semaphore machine to use Ubuntu 20.04](https://appsignal.semaphoreci.com/workflows/725495fb-3db4-4c36-a42d-753f8c2acfff). Instead, a completely different issue on a different library occurs (🙃) but if we were to sort that one out, then at least we would have upgraded the CI to a newer Ubuntu version.